kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/bytes-simplification.k

```diff
@@ -88,12 +88,11 @@ module BYTES-SIMPLIFICATION [symbolic]
       requires 0 <=Int W1 andBool 0 <=Int W2
       [simplification]
 
-    rule [buf-shift]:
-      #buf(W:Int, X <<Int Y) => #buf(W -Int (Y /Int 8), X) +Bytes #buf(Y /Int 8, 0)
-      requires 0 <=Int W andBool W <=Int 32
-       andBool 0 <=Int X andBool X <Int 2 ^Int (256 -Int Y)
-       andBool 0 <=Int Y andBool ( Y <Int W *Int 8 ) andBool Y modInt 8 ==Int 0
-      [simplification, preserves-definedness]
```

Comment:
This original rule covers the case where `W <Int 32`, as long as `X <<Int Y` fits in `W` bytes. I couldn't find a rule that could be combined with the new rule below to cover this case as well. For example, would `#buf(20, X <<Int 16)` get simplified, assuming `X <Int 2 ^Int 144`?

Reply:
I have added more lemmas to take care of this case, together with the corresponding  claim.